src/main/java/io/confluent/idesidecar/restapi/clients/SchemaErrors.java

```diff
@@ -0,0 +1,85 @@
+package io.confluent.idesidecar.restapi.clients;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+@ApplicationScoped
+public class SchemaErrors {
+
+  private final Map<String, Map<String, String>> cacheOfCaches = new ConcurrentHashMap<>();
```

Comment:
WDYT of storing the `connectionId`s in the keys of the `Map` at the first level and storing the `clusterId`s/`schemaId`s in the key of the `Map` at the second level? This could simplify this class and make methods, like `generateKey` or `generateSchemaKey`, obsolete.

Reply:
I like it!

Reply:
It would also be good to use concrete types for the first level values and maybe second level values. And we might think about using record types for the keys; definitely for the second level (since it's composite), though maybe's it's okay to just use a string for the first level as we use String Connection IDs everwhere else:

For example, maybe:
```
record ConnectionId(String id) {}
record SchemaId(String clusterId, String schemaId) {}
record Error(String message) {}
class ErrorsForConnection {
  private final ConnectionId id;
  private final Cache<SchemaId, Error> errorsBySchemaId = ...
}

private final Map<ConnectionId, ErrorsForConnection> errorsByConnectionId = ...

...
```
Nit: "connection error" might ambiguously refer to connectivity errors or errors for connections.

Using types helps in a few ways:
* Adds type safety, to avoid risking using incorrect keys or values.
* Makes the code easier to read and understand, by providing distinct terms
* If needed, we can add more to `Error`, such as maybe code, status, etc.
* Instances of record types are immutable
* We can add custom methods, if needed.

Reply:
As alluded to above, we probably want to use a Caffeine `Cache` so that the individual `Error` entries will be purged after they are added; e.g., `expireAfterWrite(Duration)`.
```
private final Cache<SchemaId, Error> errorsBySchemaId = Caffeine.newBuilder()
      .expireAfterWrite(SCHEMA_FETCH_ERROR_TTL)
      .build();
```

And we might want to actually use a Cache at the top level, so that the entry associated with a connection will be purged some time after it is last used; e.g., `expireAfterAccess(Duration)`:
```
private final Cache<ConnectionId, ErrorsForConnection> errorsByConnectionId = Caffeine.newBuilder()
      .expireAfterAccess(CONNECTION_TTL)
      .build();
```
See my next comment about handling connection lifecycle events.